.github/workflows/cypress.yml

```diff
@@ -0,0 +1,156 @@
+      image: cypress/browsers:node16.17.0-chrome106
+      options: --add-host=${{ needs.mrb-docker.outputs.domain }}:host-gateway
+      volumes:
+        - /var/run/docker.sock:/var/run/docker.sock
+    needs: mrb-docker
+    strategy:
+      fail-fast: false
+      matrix:
+        # run copies of the current job in parallel
+        containers: [ 0, 1, 2, 3, 4 ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Download MrBeamPlugin image
+        uses: actions/download-artifact@v3
+        with:
+          name: mrbeam_plugin
+          path: /tmp
+      - name: Load Docker image
+        run: |
+          apt-get --allow-releaseinfo-change update
+          apt-get install docker.io -y
+          docker load --input /tmp/image.tar
+          docker run --name mrbeam-plugin -d -p5002:5000 mrbeam_plugin
+      - run: |
+          envsubst < ./cypress/fixtures/test-data.template.json > ./cypress/fixtures/test-data.json
+          envsubst < ./cypress.config-template.js > ./cypress.config.js
+      - name: Run Cypress e2e - Chrome
+        uses: cypress-io/github-action@v4
+        with:
+          wait-on: ${{ env.MRBEAM_PLUGIN_URL }}
+          browser: chrome
+          config-file: cypress.config.js
+          command: yarn ts-node cypress/scripts/cypress-ci-run.ts
+        env:
+          TOTAL_RUNNERS: 5
+          THIS_RUNNER: ${{ matrix.containers }}
+          # Recommended: pass the GitHub token lets this action correctly
+          # determine the unique run id necessary to re-run the checks
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload JUnit reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: report_${{ matrix.containers }}
+          path: cypress/results
+
+  xray-import:
+    runs-on: ubuntu-latest
+    needs: ui-chrome-tests
+    if: always()
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Download JUnit reports
+        uses: actions/download-artifact@v3
+        with:
+          path: cypress/results
+      - name: Display structure of downloaded files
+        run: ls -R
+        working-directory: cypress/results
+      - name: Merge reports
+        run: |
+          npm install --only=dev
+          npm run merge-reports
+      - name: Import to Xray
+        id: xray
+        run: |
+          token=$(curl -H "Content-Type: application/json" -X POST --data '{ "client_id": "${{ secrets.JIRA_CLIENT_ID }}","client_secret": "${{ secrets.JIRA_CLIENT_SECRET }}" }'  https://xray.cloud.getxray.app/api/v2/authenticate| tr -d '"')
+          response=$(curl -H "Content-Type: text/xml" -X POST -H "Authorization: Bearer $token"  --data @"cypress/results/combined.xml" https://xray.cloud.getxray.app/api/v2/import/execution/junit?projectKey=SW)
+          echo "response=$response" >> $GITHUB_OUTPUT
```

Comment:
Nice oauth in 2 lines 😄 Instead of response could we name it with what we really get? I assume something like reports_response or xray_reports_response? not sure what it expects as response if I am honest.. 😓